core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -777,15 +777,17 @@ public static Jenkins getInstanceOrNull() {
      * @return The instance.
      * @throws IllegalStateException {@link Jenkins} has not been started, or was already shut down
      */
+    @SuppressFBWarnings(value="NP_NONNULL_RETURN_VIOLATION", justification="TODO delete kill switch ~2.110")
     @CLIResolver
     @Nonnull
     public static Jenkins getInstance() {
         Jenkins instance = HOLDER.getInstance();
         if (instance == null) {
-            if(SystemProperties.getBoolean(Jenkins.class.getName()+".enableExceptionOnNullInstance")) {
-                // TODO: remove that second block around 2.20 (that is: ~20 versions to battle test it)
-                // See https://github.com/jenkinsci/jenkins/pull/2297#issuecomment-216710150
-                throw new IllegalStateException("Jenkins has not been started, or was already shut down");
+            IllegalStateException ise = new IllegalStateException("Jenkins has not been started, or was already shut down");
+            if (Boolean.getBoolean(Jenkins.class.getName() + ".disableExceptionOnNullInstance")) {
+                LOGGER.log(Level.WARNING, "Perhaps you meant to use Jenkins.getInstanceOrNull?", ise);
```

Comment:
:bug:

If we are on the agent we may be unable to even load `Jenkins` so do not encourage code use that makes that possible, instead encourage use of the agent usable test api:

```
Perhaps you meant to use JenkinsJVM.isJenkinsJVM()?
```